### ISSUE TITLE
select eats all RAM and dies

### ISSUE THREAD

**Issue by jakecoffman** [jmoiron/sqlx#178]
When my server hits this line, the server starts eating all of the RAM available which eventually crashes the server: https://github.com/jakecoffman/stldevs/blob/master/web/db.go#L122

Interestingly this works locally but not on my droplet. Using go 1.2 but upgraded to 1.4 to see if that fixes it. DB is mysql, OS is ubuntu 14.04. That sql statement only ever returns 100 lines at most. 

Any ideas as to what I should try to fix this or narrow it down?

**Comment by elithrar** [comment]
Look at using EXPLAIN (https://dev.mysql.com/doc/refman/5.0/en/explain.html)
to look at the query plan. For 100 rows / 5 columns though, even with a
full heap scan I can't imagine it being very intensive even on a sub-1GB
droplet.

Also - is this a typo in the query?

`order by language, stargazers_count desc, name`

**Comment by jakecoffman** [comment]
I `explain`ed my select statement and everything looked ok, even counted to make sure it wasn't somehow returning millions of rows.

I modified my code to dump goroutines after 10 seconds of eating memory to see where it was getting stuck. For some reason Github it not allowing me to attach the dump, but a relevant line is https://github.com/jmoiron/sqlx/blob/master/reflectx/reflect.go#L282. 

I guessed that maybe the github.Repository struct's recursive nature may be getting it stuck in an infinite loop. Removing those fields seemed to fix it. It's interesting though that it works ok locally. 

Anyway, no idea if this is a bug in sqlx but it's for sure related to the environment, just thought I'd update in case someone else runs into this.

**Comment by jmoiron** [comment]
can you verify:

1. the go version on the working and non-working environment
2. the version of the github library in the working and non-working environment

A recursive structure could definitely cause some issues with the name resolver.

**Comment by jakecoffman** [comment]
- go1.4.2 windows 10 works
- go1.21 ubuntu 14.04 (DO droplet) not working
- I was using lastest sqlx on master in all environments.

**Comment by santagada** [comment]
Can we close this as its probably go 1.2 related?

**Comment by jakecoffman** [comment]
I just ran into this again using Go 1.6 using Windows 10. When scanning into this data structure it seems to get caught in a cycle: https://github.com/google/go-github/blob/master/github/repos.go#L19

**Comment by jakecoffman** [comment]
tl;dr don't use recursive structs
